gematria/datasets/pipelines/benchmark_cpu_scheduler.py

```diff
@@ -141,3 +142,19 @@ def verify(self):
     cpu_mask = list(os.sched_getaffinity(0))
     if self._cpu_mask != cpu_mask:
       raise ValueError('Expected the CPU mask to not change.')
+
+
+class BenchmarkSchedulerImplementations(Enum):
+  NoScheduling = 1
+  Default = 2
+
+
+def construct_benchmark_scheduler(
+    scheduler_type: BenchmarkSchedulerImplementations,
+) -> BenchmarkScheduler:
+  if scheduler_type == BenchmarkSchedulerImplementations.NoScheduling:
+    return NoSchedulingBenchmarkScheduler()
+  elif scheduler_type == BenchmarkSchedulerImplementations.Default:
+    return DefaultBenchmarkScheduler()
+  else:
+    raise ValueError('Unexpected Benchmark Scheduler Type.')
```

Comment:
Optional: Since 3.10, there's a new `match` keyword that covers matching one expression against multiple values, e.g.:
```py
match scheduler_type:
  case BenchmarkSchedulerImplementations.NO_SCHEDULING:
    ...
  case BenchmarkSchedulerImplementtations.DEFAULT:
    ...
  case _:
    raise ValueError(...)
```